Add methods to get and set late-bound constants.

rkimball opened the pull request:
This PR adds methods to both get and set late-bound constants.

The current approach for saving and restoring a model's late-bound constants is via the either `MoveLateBoundConstantsToFile` and then this file can be reloaded to the Executable by calling `LoadLateBoundConstantsFromFile`. These methods save the constants to a file `consts`.

I have a need to save and restore the constants in a pre-existing format. In order to write these I added `GetLateBoundConstants`  which returns a Map of all late-bound contants. To restore the constants back to the Executable `LoadLateBoundConstantsFromMap` was added.

The PR arranges the code so that both `GetLateBoundConstants` and `LoadLateBoundConstantsFromMap` are used in the existing methods to write and read late-bound constants. This means there is minimal code change, functions were mostly just rearranged to expose the two new entry points.

jwfromm commented:
Thanks @rkimball, this overall looks like a solid change. The one thing I'd like to see is the addition of some tests to make sure this works as expected.

rkimball commented:
Thank you Josh. I added some tests. Much of this new code is already tested as it is an integral part of the normal save/load. I added tests for the new API functions.

jwfromm reviewed (approved):
Thanks for adding the tests, this all LGTM!

jwfromm commented:
This is now merged, thanks @rkimball!
